tracerProvider.go

```diff
@@ -57,13 +57,23 @@ func ConfigureTracerProvider(config Config) (trace.TracerProvider, error) {
 	// Handling camelcase of provider.
 	config.Provider = strings.ToLower(config.Provider)
 	providerConfig := config.Providers[config.Provider]
+	sampleLocalTraces := config.SampleLocalTraces
 	if providerConfig == nil {
 		providerConfig = providersConfig[config.Provider]
 	}
 	if providerConfig == nil {
 		return nil, fmt.Errorf("%w for provider %s", ErrTracerProviderNotFound, config.Provider)
 	}
-	provider, err := providerConfig(config)
+
+	// Setting up trace sampler based on SampleLocalTraces value in config file
+	// Default behavior: never sample local traces
+	sampler := sdktrace.ParentBased(sdktrace.NeverSample())
+
+	if sampleLocalTraces {
+		sampler = sdktrace.ParentBased(sdktrace.AlwaysSample())
```

Comment:
I'm not sure if we should use `AlwaysSample()` even if `sampleLocalTraces` is configured to be `true`. 
An alternative option is using `TraceIDRatioBased()` (https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#TraceIDRatioBased) and only sending a percentage through.

Reply:
If we split apart the choices a bit more we can let the user decide what they want without us needing to do much & we can expand/grow later.

Replace `sampleLocalTraces` with:
```yaml
  sampling-strategy:
    parent-initiated: ignore # [ignore (default), honor]
    no-parent: never         # [never (default), always, traceidratio, ...]
    traceidratio:            # For each sampler that needs inputs they get enumerated separately.
        percentage: 15.5
```

Choosing the quietest mode (completely off) as defaults.

Until we need to add `traceidratio` we can just know it goes here.  I think @Sachin4403 may want:
```yaml
  sampling-strategy:
    parent-initiated: honor
    no-parent: always
```

We'll probably want to use:

```yaml
  sampling-strategy:
    parent-initiated: honor
    no-parent: never
```

We can add the `traceidratio` later if someone wants/needs it.

Reply:
Taking care of this in #102